test/test_symmetry.cpp

```diff
@@ -0,0 +1,39 @@
+#include <gtest/gtest.h>
+
+extern "C" {
+#include "cell.h"
+#include "symmetry.h"
+#include "utils.h"
+}
+
+TEST(test_symmetry, test_get_lattice_symmetry_layer) {
+    Cell *cell;
+    Symmetry *symmetry;
+
+    cell = NULL;
+    symmetry = NULL;
+
+    const int size = 1;
+    double lattice[3][3] = {
+        {1, 0, 0},
+        {0, 1, 0},
+        {0, 0, 1},
+    };
+    double positions[1][3] = {{0, 0, 0}};
+    const int types[1] = {0};
+    const int aperiodic_axis = 2;
+    const double symprec = 1e-5;
+    const double angle_tolerance = -1;
+
+    cell = cel_alloc_cell(size, NOSPIN);
+    cel_set_layer_cell(cell, lattice, positions, types, aperiodic_axis);
```

Comment:
```suggestion
    cell = cel_alloc_cell(size, NOSPIN);
    ASSERT_NE(cell, nullptr);
    ASSERT_EQ(spg_get_error_code(), SPGLIB_SUCCESS);
    cel_set_layer_cell(cell, lattice, positions, types, aperiodic_axis);
```
Has to be manually asserted because you cannot use `EXPECT_NO_THROW(cell = cel_alloc_cell(size, NOSPIN))`

Similar down below as well.